packages/react-openapi/src/OpenAPISchema.tsx

```diff
@@ -190,12 +192,13 @@ function OpenAPISchemaAlternative(props: {
 }) {
     const { schema, circularRefs, context } = props;
     const id = useId();
+    const subProperties = getSchemaProperties(schema);
 
     return (
         <OpenAPISchemaProperties
             id={id}
-            properties={getSchemaProperties(schema) ?? []}
-            circularRefs={new Map(circularRefs).set(schema, id)}
+            properties={subProperties ?? [{ schema }]}
+            circularRefs={subProperties ? new Map(circularRefs).set(schema, id) : circularRefs}
```

Comment:
This feels smelly that we are calling new here directly in the prop. It would be cleaner to pull this up and out of the JSX I think.

Reply:
Why? it's done in a bunch of places here, it makes sure we create a new reference without impacting the other ones used in the react tree (map as an immutable map without using a library for this).

Pulling this up will make it even less clear imo as you'll lose track of where it's used and why we need a new.

Reply:
The thinking is that it should be more clear because you'll need to name this logic. It's also fairly un-Reacty. 
this function here was named so was a bit easier to understand immediately: `getSchemaProperties(schema)`.
Maybe a style difference? not sure but it seems less readable for me as I have to stop and look closely at it to be sure it's not gonna cause an optimization issue.